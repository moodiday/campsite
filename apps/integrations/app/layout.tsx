// Next.js automatically generates this file if it doesn't exist
export const metadata = {
  title: 'Integrations',
  description: 'Integrations'
}

export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <html lang='en'>
      <body>{children}</body>
    </html>
  )
}
