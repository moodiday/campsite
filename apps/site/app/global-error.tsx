'use client'

import { useEffect } from 'react'
import * as Sentry from '@sentry/nextjs'
import NextError from 'next/error'

/**
 * @see https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#react-render-errors-in-app-router
 */
export default function GlobalError({ error }: { error: Error & { digest?: string } }) {
  useEffect(() => {
    Sentry.captureException(error)
  }, [error])

  return (
    <html>
      <body>
        <NextError statusCode={500} />
      </body>
    </html>
  )
}
