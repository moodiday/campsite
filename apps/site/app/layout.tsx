import { Inter } from 'next/font/google'

import '@campsite/ui/src/styles/global.css'
import '@campsite/ui/src/styles/code.css'
import 'styles/global.css'

import { Metadata } from 'next'

import { DEFAULT_SEO } from '@campsite/config'

import Providers from '@/components/Providers'

const inter = Inter({
  subsets: ['latin'],
  variable: '--font-inter'
})

export const metadata: Metadata = DEFAULT_SEO

export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <html lang='en' suppressHydrationWarning className={inter.variable}>
      <body className='bg-primary dark:bg-neutral-950'>
        <span className='sr-only'>
          <a href='#main'>Skip to content</a>
          <a href='#list'>jump to list</a>
        </span>

        <Providers router='app'>{children}</Providers>
      </body>
    </html>
  )
}
