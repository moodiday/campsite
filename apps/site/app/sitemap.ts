import { getGlossaries } from 'app/glossary/_cms/getGlossaries'
import { MetadataRoute } from 'next'

import { SITE_URL } from '@campsite/config'

import { getBlogs } from '@/lib/blog'
import { getChangelog } from '@/lib/changelog'

export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
  const changelogs = await getChangelog()
  const glossaries = await getGlossaries()

  return [
    {
      url: SITE_URL,
      lastModified: new Date()
    },
    {
      url: `${SITE_URL}/pricing`,
      lastModified: new Date()
    },
    {
      url: `${SITE_URL}/contact`,
      lastModified: new Date()
    },
    {
      url: `${SITE_URL}/privacy`,
      lastModified: new Date()
    },
    {
      url: `${SITE_URL}/terms`,
      lastModified: new Date()
    },
    {
      url: `${SITE_URL}/dpa`,
      lastModified: new Date()
    },
    {
      url: `${SITE_URL}/changelog`,
      lastModified: new Date()
    },
    {
      url: `${SITE_URL}/blog`,
      lastModified: new Date()
    },
    ...getBlogs().map((blog) => ({
      url: `${SITE_URL}/blog/${blog.slug}`,
      lastModified: new Date(blog.metadata.publishedAt).toISOString()
    })),
    ...changelogs.map((changelog) => ({
      url: `${SITE_URL}/changelog/${changelog.data.slug}`,
      lastModified: changelog.data.isoDate
    })),
    {
      url: `${SITE_URL}/glossary/slack`,
      lastModified: new Date()
    },
    ...glossaries.map((glossary) => ({
      url: `${SITE_URL}/glossary/${glossary.category}/${glossary.slug.current}`,
      lastModified: glossary._updatedAt
    })),
    {
      url: `${SITE_URL}/resources/linear-keyboard-shortcuts`,
      lastModified: new Date()
    },
    {
      url: `${SITE_URL}/switch-from-slack`,
      lastModified: new Date()
    }
  ]
}
