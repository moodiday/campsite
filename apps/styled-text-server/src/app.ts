import * as Sentry from '@sentry/node'
import bodyParser from 'body-parser'
import express from 'express'
import morgan from 'morgan'

import { htmlToSlack } from './html'
import { markdownToHtml } from './markdown'

export const app: express.Express = express()

if (process.env.NODE_ENV === 'production') {
  Sentry.init({
    dsn: 'https://cace67e62da1a116fd2fa2a30a3cd1d9@o1244295.ingest.us.sentry.io/4506985116401664',
    debug: process.env.NODE_ENV !== 'production',
    environment: process.env.NODE_ENV ?? 'development',
    tracesSampleRate: 0
  })
}

console.log('NODE_ENV', process.env.NODE_ENV)

// The request handler must be the first middleware on the app
app.use(Sentry.Handlers.requestHandler())

app.use(morgan('combined'))
app.use(bodyParser.text({ limit: '5mb' }))
app.use(bodyParser.urlencoded({ limit: '5mb', extended: true }))
app.use((req, res, next) => {
  if (req.header('Authorization') != `Bearer ${process.env.AUTHTOKEN}`) {
    return res.status(401).json({ message: 'invalid token' })
  }

  next()
})

app.post('/html_to_slack', bodyParser.json(), (req, res) => {
  if (!req.is('*/json')) {
    return res.status(415).json({ message: 'request must be application/json' })
  }

  const { html } = req.body

  const blocks = htmlToSlack(html)

  res.json(blocks)
})

app.post('/markdown_to_html', bodyParser.json(), (req, res) => {
  if (!req.is('*/json')) {
    return res.status(415).json({ message: 'request must be application/json' })
  }

  const { markdown, editor } = req.body

  if (!markdown) {
    return res.status(400).json({ message: 'markdown and editor are required' })
  }

  if (typeof markdown !== 'string') {
    return res.status(400).json({ message: 'markdown must be a string' })
  }

  res.json({ html: markdownToHtml(markdown, editor) })
})

// The error handler must be before any other error middleware and after all controllers
app.use(
  Sentry.Handlers.errorHandler({
    shouldHandleError(error) {
      return !error.status || parseInt(`${error.status}`) >= 400
    }
  })
)
