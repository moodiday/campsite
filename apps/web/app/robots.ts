import { MetadataRoute } from 'next'

import { SITE_URL } from '@campsite/config'

export default function robots(): MetadataRoute.Robots {
  // Disallow crawling all pages under `app.campsite` and point the sitemap to `www.campsite`
  return {
    rules: {
      userAgent: '*',
      disallow: '/'
    },
    sitemap: `${SITE_URL}/sitemap.xml`
  }
}
