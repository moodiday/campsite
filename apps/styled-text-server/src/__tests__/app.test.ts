/* eslint-disable max-lines */
import supertest from 'supertest'
import { describe, expect, it } from 'vitest'

import { app } from '../app'

const KITCHEN_SYNC_MARKDOWN = `
# Heading 1
## Heading 2
### Heading 3
#### Heading 4
##### Heading 5
###### Heading 6

<!-- this is a comment -->

This is a paragraph with **bold** and italics.

- Bullet one
- Bullet two

1. Number one
2. Number two

\`\`\`
const foo = "bar"
\`\`\`

Hard break
Soft break

---

![CleanShot 2024-03-22 at 16 42 23@2x](https://github.com/campsite/campsite/assets/739696/49b398b1-8c03-4255-a759-21b8b53a3f5d)

| Header | Header | Header |
|--------|--------|--------|
| Cell | Cell | Cell |
| Cell | Cell | Cell | 

> Ullamco eiusmod laborum minim nulla adipisicing incididunt occaecat consequat non ipsum ex qui excepteur culpa.

And [here](https://linear.app/campsite/issue/CAM-6845/normalize-notes) is a link. With a \`inline code\` mark.
`

describe('app', () => {
  describe('POST /html_to_slack', () => {
    it('accepts html and returns Slack blocks', async () => {
      const html = `<h1>My markdown</h1>

<p>It has <b>bold</b> and <i>italics</i> and a <a href="https://campsite.com"><b>bold link</b></a>.</p>

<p>It has multiple paragraphs.</p>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: '*My markdown*',
            verbatim: true
          }
        },
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: 'It has *bold* and _italics_ and a <https://campsite.com|*bold link*>.',
            verbatim: true
          }
        },
        {
          type: 'section',
          text: {
            text: 'It has multiple paragraphs.',
            type: 'mrkdwn',
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('converts task lists into emoji', async () => {
      const html = `<ul class="task-list" data-type="taskList">
      <li class="task-item" data-checked="false" data-type="taskItem">
        <label><input type="checkbox"><span></span></label>
        <div><p>Unchecked</p></div>
      </li>
      <li class="task-item" data-checked="true" data-type="taskItem">
        <label><input type="checkbox" checked="checked"><span></span></label>
        <div><p>Checked</p></div>
      </li>
    </ul>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: `• :white_medium_small_square: Unchecked
• ✓ Checked`,
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('converts task lists with empty checked value into emoji', async () => {
      const html = `<ul class="task-list" data-type="taskList">
      <li class="task-item" data-checked="false" data-type="taskItem">
        <label><input type="checkbox"><span></span></label>
        <div><p>Unchecked</p></div>
      </li>
      <li class="task-item" data-checked="true" data-type="taskItem">
        <label><input type="checkbox" checked><span></span></label>
        <div><p>Checked</p></div>
      </li>
    </ul>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: `• :white_medium_small_square: Unchecked
• ✓ Checked`,
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('strips empty sections', async () => {
      const html = `<p>Foo bar</p>
    <p><br><br></p>
    <p>Cat dog<br><br>Boy howdy</p>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: `Foo bar`,
            verbatim: true
          }
        },
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: `Cat dog\n\nBoy howdy`,
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('retains blockquotes and wraps in section', async () => {
      const html = `<blockquote>Foo bar</blockquote>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            // extra newline added by jsx-slack and trimmed by slack
            text: `&gt; Foo bar
&gt; `,
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('converts <img> tags to images', async () => {
      const html = `<img src="https://campsite.com/image.png" alt="alt text" />`

      const expectedBlocks = [
        {
          type: 'image',
          image_url: 'https://campsite.com/image.png',
          alt_text: 'alt text'
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('converts link unfurls into urls', async () => {
      const html = `<link-unfurl href="https://campsite.com"></link-unfurl>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: 'https://campsite.com',
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('parses inline attachments', async () => {
      const html = `<post-attachment id="1scvs2jokefh" file_type="" width="0" height="0"></post-attachment><post-attachment id="1scvs2jokefh" file_type="image/png" width="0" height="0"></post-attachment>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: '_📄 Open this post on Campsite to view this attachment._',
            verbatim: true
          }
        },
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: '_🖼️ Open this post on Campsite to view this attachment._',
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('parses media galleries', async () => {
      const html = `<media-gallery><media-gallery-item id="foo"></media-gallery-item></media-gallery>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: '_🖼️ Open this post on Campsite to view attachments._',
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('strips extremely long links', async () => {
      const html = `<p>my <a href="${'a'.repeat(3000)}">link</a></p>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: 'my link',
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('splits long unordered lists into separate blocks', async () => {
      const html = `
        <ul>
          <li>
            <p>First item <a href="${'a'.repeat(1990)}">a</a></p>
            <ul>
              <li>First sub-item</li>
              <li>Second sub-item</li>
            </ul>
          </li>
          <li>
            <p>Second item</p>
            <ul>
              <li>First sub-item</li>
              <li>Second sub-item</li>
            </ul>
          </li>
        </ul>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            // eslint-disable-next-line no-irregular-whitespace
            text: `• First item <${'a'.repeat(1990)}|a>\n  ◦ First sub-item\n  ◦ Second sub-item`,
            verbatim: true
          }
        },
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: '• Second item\n  ◦ First sub-item\n  ◦ Second sub-item',
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('splits long ordered lists into separate blocks', async () => {
      const html = `
        <ol>
          <li>
            <p>First item ${'a'.repeat(2000)}</p>
            <ol>
              <li>First sub-item</li>
              <li>Second sub-item</li>
            </ul>
          </li>
          <li>
            <p>Second item</p>
            <ol>
              <li>First sub-item</li>
              <li>Second sub-item</li>
            </ul>
          </li>
        </ol>`

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            // eslint-disable-next-line no-irregular-whitespace
            text: `1. First item ${'a'.repeat(2000)}\n   1. First sub-item\n   2. Second sub-item`,
            verbatim: true
          }
        },
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: '2. Second item\n   1. First sub-item\n   2. Second sub-item',
            verbatim: true
          }
        }
      ]

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('replaces too long sections', async () => {
      const html = '<p>' + 'a'.repeat(3001) + '</p>'

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: '_Removed a section too long to display in Slack._',
            verbatim: true
          }
        }
      ]

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('replaces custom reactions with names', async () => {
      const html =
        '<p>yo <img data-type="reaction" src="https://campsite.imgix.net/custom-reactions-packs/blobs/blob-smile-happy.png" alt="blob-smile-happy" draggable="false" data-id="dvyazibgmy8w" data-name="blob-smile-happy"></p>'

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            type: 'mrkdwn',
            text: 'yo :blob-smile-happy:',
            verbatim: true
          }
        }
      ]

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('handles <details> tags', async () => {
      const html = '<details><summary>Summary</summary><div><p>Details</p></div></details>'

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            text: '*Summary*',
            type: 'mrkdwn',
            verbatim: true
          }
        },
        {
          type: 'section',
          text: {
            text: 'Details',
            type: 'mrkdwn',
            verbatim: true
          }
        }
      ]

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })

    it('handles <kbd> tags', async () => {
      const html = '<p>Shortcut: <kbd>Mod</kbd> <kbd>F</kbd></p>'

      const res = await supertest(app)
        .post('/html_to_slack')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ html })

      const expectedBlocks = [
        {
          type: 'section',
          text: {
            text: 'Shortcut: `Mod` `F`',
            type: 'mrkdwn',
            verbatim: true
          }
        }
      ]

      expect(res.statusCode).toEqual(200)
      expect(res.body).toEqual(expectedBlocks)
    })
  })
  describe('POST /markdown_to_html', () => {
    it('errors on bad editor', async () => {
      const res = await supertest(app)
        .post('/markdown_to_html')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ markdown: 'foo **bar**', editor: 'wrong' })

      expect(res.statusCode).toEqual(500)
    })

    it('errors on non-string markdown', async () => {
      const res = await supertest(app)
        .post('/markdown_to_html')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ markdown: 2, editor: 'markdown' })

      expect(res.statusCode).toEqual(400)
    })

    it('it returns note html', async () => {
      const res = await supertest(app)
        .post('/markdown_to_html')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ markdown: KITCHEN_SYNC_MARKDOWN, editor: 'note' })

      expect(res.statusCode).toEqual(200)
      expect(res.body.html).toMatchSnapshot()
    })

    it('it returns markdown html', async () => {
      const res = await supertest(app)
        .post('/markdown_to_html')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ markdown: KITCHEN_SYNC_MARKDOWN, editor: 'markdown' })

      expect(res.statusCode).toEqual(200)
      expect(res.body.html).toMatchSnapshot()
    })

    it('it returns chat html', async () => {
      const res = await supertest(app)
        .post('/markdown_to_html')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ markdown: KITCHEN_SYNC_MARKDOWN, editor: 'chat' })

      expect(res.statusCode).toEqual(200)
      expect(res.body.html).toMatchSnapshot()
    })

    it('it retains input html', async () => {
      const input =
        'This is **bold** and this is a mention <span data-type="mention" data-id="abcdefabcdef" data-label="User Name" data-username="username">@User Name</span>.'

      const res = await supertest(app)
        .post('/markdown_to_html')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ markdown: input, editor: 'markdown' })

      expect(res.statusCode).toEqual(200)
      expect(res.body.html).toMatchSnapshot()
    })

    it('it trims whitespace', async () => {
      const input = `


# Foo bar

Cat dog                              
`

      const res = await supertest(app)
        .post('/markdown_to_html')
        .set('Content-Type', 'application/json')
        .set('Authorization', `Bearer ${process.env.AUTHTOKEN}`)
        .send({ markdown: input, editor: 'markdown' })

      expect(res.statusCode).toEqual(200)
      expect(res.body.html).toMatchSnapshot()
    })
  })
})
