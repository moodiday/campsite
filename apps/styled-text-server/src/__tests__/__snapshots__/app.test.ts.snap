// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app > POST /markdown_to_html > it retains input html 1`] = `"<p>This is <strong>bold</strong> and this is a mention <span class="mention" data-type="mention" data-id="abcdefabcdef" data-label="User Name" data-role="member" data-username="username">@User Name</span>.</p>"`;

exports[`app > POST /markdown_to_html > it returns chat html 1`] = `"<p>Heading 1</p><p>Heading 2</p><p>Heading 3</p><p>Heading 4</p><p>Heading 5</p><p>Heading 6</p><p>This is a paragraph with <strong>bold</strong> and italics.</p><p>Bullet one</p><p>Bullet two</p><p>Number one</p><p>Number two</p><p>const foo = "bar"</p><p>Hard breakSoft break</p><p>HeaderHeaderHeaderCellCellCellCellCellCell</p><p>And <a target="_blank" rel="noopener noreferrer nofollow" class="prose-link" href="https://linear.app/campsite/issue/CAM-6845/normalize-notes"><span>here</span></a> is a link. With a <code spellcheck="false">inline code</code> mark.</p>"`;

exports[`app > POST /markdown_to_html > it returns markdown html 1`] = `"<h1>Heading 1</h1><h2>Heading 2</h2><h3>Heading 3</h3><h4>Heading 4</h4><h5>Heading 5</h5><h6>Heading 6</h6><p>This is a paragraph with <strong>bold</strong> and italics.</p><ul><li><p>Bullet one</p></li><li><p>Bullet two</p></li></ul><ol><li><p>Number one</p></li><li><p>Number two</p></li></ol><pre spellcheck="false"><code>const foo = "bar"</code></pre><p>Hard break<br>Soft break</p><div data-hr-wrapper="true"><hr></div><p><a target="_blank" rel="noopener noreferrer nofollow" class="prose-link" href="https://github.com/campsite/campsite/assets/739696/49b398b1-8c03-4255-a759-21b8b53a3f5d"><span>CleanShot 2024-03-22 at 16 42 23@2x</span></a></p><p>HeaderHeaderHeaderCellCellCellCellCellCell</p><blockquote><p>Ullamco eiusmod laborum minim nulla adipisicing incididunt occaecat consequat non ipsum ex qui excepteur culpa.</p></blockquote><p>And <a target="_blank" rel="noopener noreferrer nofollow" class="prose-link" href="https://linear.app/campsite/issue/CAM-6845/normalize-notes"><span>here</span></a> is a link. With a <code spellcheck="false">inline code</code> mark.</p>"`;

exports[`app > POST /markdown_to_html > it returns note html 1`] = `"<h1>Heading 1</h1><h2>Heading 2</h2><h3>Heading 3</h3><h4>Heading 4</h4><h5>Heading 5</h5><h6>Heading 6</h6><p>This is a paragraph with <strong>bold</strong> and italics.</p><ul><li><p>Bullet one</p></li><li><p>Bullet two</p></li></ul><ol><li><p>Number one</p></li><li><p>Number two</p></li></ol><pre spellcheck="false"><code>const foo = "bar"</code></pre><p>Hard break<br>Soft break</p><div data-hr-wrapper="true"><hr></div><p><a target="_blank" rel="noopener noreferrer nofollow" class="prose-link" href="https://github.com/campsite/campsite/assets/739696/49b398b1-8c03-4255-a759-21b8b53a3f5d"><span>CleanShot 2024-03-22 at 16 42 23@2x</span></a></p><p>HeaderHeaderHeaderCellCellCellCellCellCell</p><blockquote><p>Ullamco eiusmod laborum minim nulla adipisicing incididunt occaecat consequat non ipsum ex qui excepteur culpa.</p></blockquote><p>And <a target="_blank" rel="noopener noreferrer nofollow" class="prose-link" href="https://linear.app/campsite/issue/CAM-6845/normalize-notes"><span>here</span></a> is a link. With a <code spellcheck="false">inline code</code> mark.</p>"`;

exports[`app > POST /markdown_to_html > it trims whitespace 1`] = `"<h1>Foo bar</h1><p>Cat dog</p>"`;
